### PR TITLE
Feature 114 single reply view

### LINKED ISSUE #114: Single reply view
As a user I'd like to see single reply as filled form in it's own view

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #114

### CODE REVIEW

File: src/components/screens/forms/form-reply-screen.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
+import React, { FC, useContext, useEffect, useState } from "react";
+import strings from "localization/strings";
+import { Metaform, MetaformFieldType, Reply } from "generated/client";
+import { FieldValue, FileFieldValue } from "metaform-react/types";
+
+import MetaformUtils from "utils/metaform-utils";
+import Form from "components/generic/form";
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:7]:
```suggestion
import { FieldValue, FileFieldValue } from "metaform-react/types";
import MetaformUtils from "utils/metaform-utils";
import Form from "components/generic/form";
```

File: src/components/screens/forms/form-reply-screen.tsx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+ */
+const ReplyScreen: FC = () => {
+  const errorContext = useContext(ErrorContext);
+
+  const [ loading, setLoading ] = useState(false);
+
+  const [ metaform, setMetaform ] = useState<Metaform>(MetaformUtils.jsonToMetaform({}));
+  const [ formValues, setFormValues ] = useState<Dictionary<FieldValue>>({});
+  const [ reply, setReply ] = useState<Reply>();
+  const [ replySavedVisible, setReplySavedVisible ] = useState(false);
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:32]:
```suggestion
  const [ loading, setLoading ] = useState(false);
  const [ metaform, setMetaform ] = useState<Metaform>(MetaformUtils.jsonToMetaform({}));
  const [ formValues, setFormValues ] = useState<Dictionary<FieldValue>>({});
  const [ reply, setReply ] = useState<Reply>();
  const [ replySavedVisible, setReplySavedVisible ] = useState(false);
```

File: src/components/screens/forms/form-reply-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+  }
+
+  /**
+   * Method for getting field value
+   *
+   * @param fieldName field name
+   */
+  const getFieldValue = (fieldName: string): FieldValue => {
+    return formValues[fieldName] || null;
+  };
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:56]:
```suggestion
  const getFieldValue = (fieldName: string): FieldValue => formValues[fieldName] || null;
```

File: src/components/screens/forms/form-reply-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+  /**
+   * Method for saving reply
+   */
+  const onSave = async () => {
+    if (!metaform || !metaform.id || !reply || !reply.id) {
+      return;
+    }
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:64]:
```suggestion
    if (!metaform?.id || !reply?.id) {
      return;
    }
```

File: src/components/screens/forms/form-reply-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        setFieldValue={ setFieldValue }
+        onSubmit={ onSave }
+      />
+    );
+  };
+
+  /**
+   * View setup
+   */
+  const setup = async () => {
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:162]: Rename this function to `loadData`

File: src/components/screens/forms/form-reply-screen.tsx [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+      return;
+    }
+
+    const updatedReplyData = { ...reply.data, status: event.target.value };
+    const updatedReply = { ...reply, data: updatedReplyData as any };
+    setReply(updatedReply);
+  };
+
+  const statusOptions = Object.keys(ReplyStatus).map(status => { return status; });
+  console.log(LocalizationUtils.getLocalizedStatusOfReply(statusOptions[0]));
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:206]: remove logs?

File: src/components/screens/forms/form-reply-screen.tsx [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+            );
+          })
+        }
+      </StatusSelector>
+    );
+  };
+
+  useEffect(() => {
+    setup();
+  }, []);
```
Comment (Chi-Chu319) [inline src/components/screens/forms/form-reply-screen.tsx:237]: move all the load functions to the top followed by the use effect hook calls

File: src/components/screens/forms/form-reply-screen.tsx (hunk 6 above)
Comment (EVP1989) [reply src/components/screens/forms/form-reply-screen.tsx:202]: Changing to comment TODO since the localization is not working yet

**Pull request merged**